app/jobs/preassembly_job.rb

```diff
@@ -13,6 +13,8 @@ def perform(job_run)
     else
       job_run.preassembly_completed
     end
+    # To avoid a possible race condition, run JobrunCompleteJob.
+    JobRunCompleteJob.perform_later(job_run)
```

Comment:
Thanks for leaving a comment here. It's not immediately obvious to me how it prevents race conditions, but at least there is a comment explaining why it is here.

Reply:
i'm curious about the possible race condition too

Reply:
Thank you for your curiosity. Led me to realize that I didn't quite have the race condition covered correctly. See https://github.com/sul-dlss/pre-assembly/pull/1279 for a fix.